### PR TITLE
Add mock autoscaling

### PR DESCRIPTION
**Pull request by andrewsykim**
Part of https://github.com/kubernetes/kops/issues/1761 (testing rolling-update code). 

So far I've only added mock implementation for functions that I think will be needed for unit testing later. I may end up adding more as I go.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1824)

### CODE REVIEW

**Comment by justinsb** [comment]
@k8s-bot ok to test

(Although I think travis is unhappy)

File: cloudmock/aws/mockautoscaling/api.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
```
Comment (chrislovecnm) [inline cloudmock/aws/mockautoscaling/api.go]: Should we have the submodule changes?

Comment (andrewsykim) [inline cloudmock/aws/mockautoscaling/api.go]: sorry, must have been an accident

Comment (andrewsykim) [inline cloudmock/aws/mockautoscaling/api.go]: done

**Comment by andrewsykim** [comment]
Not sure why travisci check is failing, running 
```
hack/verify-packages.sh
```
on my local machine passes 🤔

**Comment by chrislovecnm** [comment]
> hack/.packages is not in up to date. Please run:
  go list k8s.io/kops/... | grep -v vendor > hack/.packages
make: *** [verify-packages] Error 1

You update it fully?  My be a bug in the hack scritpt

File: hack/.packages [hunk 2]
```diff
@@ -13,7 +14,6 @@ k8s.io/kops/dns-controller/pkg/util
 k8s.io/kops/dns-controller/pkg/watchers
 k8s.io/kops/examples/kops-api-example
 k8s.io/kops/federation
-k8s.io/kops/federation/model
```
Comment (chrislovecnm) [inline hack/.packages]: did you mean to remove this line?

Comment (andrewsykim) [inline hack/.packages]: That was the result I got from running 
```
go list k8s.io/kops/... | grep -v vendor > hack/.packages"
```
and I did double check, that directory is not a go package https://github.com/kubernetes/kops/tree/master/federation/model

**Comment by andrewsykim** [comment]
@chrislovecnm I fixed the travis ci issue. The problem is that the go package [here](https://github.com/kubernetes/kops/tree/master/federation/model) is generated somewhere in the CI pipeline and I ran
```
go list k8s.io/kops/... | grep -v vendor > hack/.packages
```
on a clean slate. If I run 
```
make ci
```
I was able to catch the diff. I can see this being a source of confusion for others down the road but I can't think of any good ways to fix it at the moment.

**Comment by justinsb** [comment]
Thanks @andrewsykim ... the package list seemed like a good idea at the time, but we're not using it that much (yet).  I think we can wait and see on it.

I think adding a package _should_ be a little hard ;-)  Though every time I hit it myself I am also annoyed by it!

**Pull request merged**